packages/apify/src/actor.ts

```diff
@@ -512,15 +514,21 @@ export class Actor<Data extends Dictionary = Dictionary> {
      * For more information, see the [Actor Runs](https://docs.apify.com/api/v2#/reference/actor-runs/) API endpoints.
      * @ignore
      */
-    async setStatusMessage(statusMessage: string): Promise<ClientActorRun> {
+    async setStatusMessage(statusMessage: string, options?: { isStatusMessageTerminal?: boolean }): Promise<ClientActorRun> {
+        const { isStatusMessageTerminal } = options || {};
         ow(statusMessage, ow.string);
+        ow(isStatusMessageTerminal, ow.optional.boolean);
 
         const runId = this.config.get('actorRunId')!;
         if (!runId) {
+            if (!this.isAtHome()) {
+                log.info(`Setting status message to ${statusMessage}...`);
+                return;
```

Comment:
this might need some polishment, either return some empty object like response, or ideally rework it so it can pass the call to current storage implementation